File: 0812-largest-triangle-area/0812-largest-triangle-area.java
class Solution {
    public double largestTriangleArea(int[][] points) {
        /*
            -- How does one find the area of the triangle?
            -- If it is a right-angled triangle -- 0.5 * base * height.
            -- Or heron's formula if the sides are given.
            -- Or when coordinates are given -- 0.5 * {x1(y2 - y3) + x2(y3 - y1) + x3(y1 - y2)}.
            -- so we will create a function called double areaOfTriangle (int[] one, int[] two, int[] three).

            -- since there are only max 50 points, brute force can be used.

            -- Triple nested loop.
        */

        double maxArea = 0;
        double area = 0;

        for (int i = 0; i < points.length; i++){
            for (int j = i + 1; j < points.length; j++){
                for (int k = j + 1; k < points.length; k++){
                    area = areaOfTriangle(points[i], points[j], points[k]);
                    maxArea = Math.max(maxArea, area);
                }
            }
        }
        return maxArea;

    }

    private double areaOfTriangle (int[] one, int[] two, int[] three){
        return 0.5 * Math.abs((one[0]*(two[1] - three[1]) + two[0]*(three[1] - one[1]) + three[0]*(one[1] - two[1])));
    }
}

File: 0812-largest-triangle-area/0812-largest-triangle-area.py
class Solution:
    def largestTriangleArea(self, points: list[list[int]]) -> float:
        """
            -- How does one find the area of the triangle?
            -- If it is a right-angled triangle -- 0.5 * base * height.
            -- Or Heron's formula if the sides are given.
            -- Or when coordinates are given -- 0.5 * {x1(y2 - y3) + x2(y3 - y1) + x3(y1 - y2)}.
            -- so we will create a function called area_of_triangle(one, two, three).

            -- since there are only max 50 points, brute force can be used.

            -- Triple nested loop.
        """

        def area_of_triangle(one: list[int], two: list[int], three: list[int]) -> float:
            return 0.5 * abs(
                one[0] * (two[1] - three[1]) +
                two[0] * (three[1] - one[1]) +
                three[0] * (one[1] - two[1])
            )

        max_area = 0.0
        n = len(points)

        for i in range(n):
            for j in range(i + 1, n):
                for k in range(j + 1, n):
                    area = area_of_triangle(points[i], points[j], points[k])
                    max_area = max(max_area, area)

        return max_area
